### PR TITLE
apps/builtin: Unify builtin header files

### PR DESCRIPTION
**Pull request by jeongarmy**
There are 3 header files for builtin app info.
Some information of them are duplicated so they can be unified.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [af537831e75ecbda6420a9554c30b5265173a00e] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [af537831e75ecbda6420a9554c30b5265173a00e] - Code Rule Check Result:
os/drivers/task_manager/task_manager_drv.c:154: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'

File: framework/src/task_manager/task_manager.c [hunk 1]
```diff
@@ -62,10 +63,10 @@ typedef struct task_list_data_s task_list_data_t;
 
 static int builtin_cnt;
 static mqd_t g_tm_recv_mqfd;
-extern const task_builtin_list_t tm_builtin_lists[];
 /* [0] : tm_gid, [1] : the address of task list data */
 static task_list_t tm_task_list[CONFIG_TASK_MANAGER_MAX_TASKS];
-extern int get_tm_builtin_cnt(void);
+extern const builtin_info_t builtin_list[];
+extern int get_builtin_list_cnt(void);
```
Comment (sunghan-chang) [inline framework/src/task_manager/task_manager.c]: Please put these two prototype in builtin.h

File: os/drivers/task_manager/task_manager_drv.c [hunk 2]
```diff
@@ -149,7 +150,8 @@ static int taskmgt_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
 		ret = OK;
 		break;
 	case TMIOC_START:
-		task_info = (task_builtin_list_t *)arg;
+		task_info = (builtin_info_t *)arg;
+		tmdbg("%s %d %d %p\n", task_info->name, task_info->priority, task_info->stacksize, task_info->entry );
```
Comment (sunghan-chang) [inline os/drivers/task_manager/task_manager_drv.c]: coding rule

**Comment by seinfra** [comment]
Target : [003beaf079b1e325f3a11dc43244f5ea0fcc3dbe] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [003beaf079b1e325f3a11dc43244f5ea0fcc3dbe] - Code Rule Check OK.

File: apps/include/builtin.h [hunk 3]
```diff
@@ -19,6 +19,20 @@
...
+	const char *name;
+	const TASK_CALLBACK entry;
+	const int exectype;
+	const int priority;
+	const int stacksize;
+};
+typedef struct builtin_info_s builtin_info_t;
+
+extern const builtin_info_t builtin_list[];
+extern int get_builtin_list_cnt(void);
```
Comment (jeongchanKim) [inline apps/include/builtin.h]: remove extern

File: apps/include/shell/tash.h [hunk 4]
```diff
@@ -28,6 +28,7 @@
 #define __APPS_INCLUDE_SHELL_TASH_H
 
 #include <tinyara/config.h>
+#include <apps/builtin.h>
```
Comment (jeongchanKim) [inline apps/include/shell/tash.h]: I think #ifdef CONFIG_BUILTIN_APPS is needed

File: apps/shell/tash_command.c [hunk 5]
```diff
@@ -25,6 +25,7 @@
 #include <sys/types.h>
 #include <sys/boardctl.h>
 #include <apps/shell/tash.h>
+#include <apps/builtin.h>
```
Comment (jeongchanKim) [inline apps/shell/tash_command.c]: check ifdef condition

File: apps/include/shell/tash.h (hunk 4 above)
Comment (jeongarmy) [reply apps/include/shell/tash.h]: I think this is needed because of line 42.
The structure of built-in is also used for tash command.

**Comment by seinfra** [comment]
Target : [b0e16d1e987af56fb9f5cf34fd0ab3aa00761d65] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [b0e16d1e987af56fb9f5cf34fd0ab3aa00761d65] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [947b7049ebeb6cd4bb4fff9b1d42ea0ea7aa5530] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [947b7049ebeb6cd4bb4fff9b1d42ea0ea7aa5530] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [0881796e11b43532995e11d7645d70c94387d966] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [0881796e11b43532995e11d7645d70c94387d966] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [96594bd6d876315ffa139615daa9cda1359a7f96] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [96594bd6d876315ffa139615daa9cda1359a7f96] - Code Rule Check OK.

**Pull request merged**